### ISSUE TITLE
PM2 global error caught

### ISSUE THREAD

**Issue by extensionsapp** [Unitech/pm2#2837]
**What's going wrong?**
Stop all pm2 sites.

**How could we reproduce this issue?**
Watch the logs.

## Supporting information

```
PM2 version: 2.4.4
Node version: v6.10.2
Debian 8 x64
```

logs
```
2017-04-17 11:01:28: Change detected on path restart.server for app app1 - restarting
2017-04-17 11:01:28: Stopping app:app1 id:5
2017-04-17 11:01:28: App [app1] with id [5] and pid [18251], exited with code [0] via signal [SIGINT]
2017-04-17 11:01:28: pid=18251 msg=process killed
2017-04-17 11:01:28: Starting execution sequence in -fork mode- for app name:app1 id:5
2017-04-17 11:01:28: App name:app1 id:5 online
2017-04-17 11:01:59: Change detected on path restart.server for app app2 - restarting
2017-04-17 11:01:59: Stopping app:app2 id:9
2017-04-17 11:01:59: App [app2] with id [9] and pid [18279], exited with code [0] via signal [SIGINT]
2017-04-17 11:01:59: pid=18279 msg=process killed
2017-04-17 11:01:59: Starting execution sequence in -fork mode- for app name:app2 id:9
2017-04-17 11:01:59: App name:app2 id:9 online
2017-04-17 11:02:01: Change detected on path restart.server for app app3 - restarting
2017-04-17 11:02:01: Stopping app:app3 id:8
2017-04-17 11:02:01: App [app3] with id [8] and pid [18022], exited with code [0] via signal [SIGINT]
2017-04-17 11:02:01: pid=18022 msg=process killed
2017-04-17 11:02:01: Starting execution sequence in -fork mode- for app name:app3 id:8
2017-04-17 11:02:01: App name:app3 id:8 online
2017-04-17 11:50:41: Change detected on path core for app app4 - restarting
2017-04-17 11:50:41: Stopping app:app4 id:7
2017-04-17 11:50:42: ===============================================================================
2017-04-17 11:50:42: --- PM2 global error caught ---------------------------------------------------
2017-04-17 11:50:42: Time                 : Mon Apr 17 2017 11:50:42 GMT+0300 (MSK)
2017-04-17 11:50:42: spawn ENOMEM
2017-04-17 11:50:42: Error: spawn ENOMEM
    at exports._errnoException (util.js:1018:11)
    at ChildProcess.spawn (internal/child_process.js:319:11)
    at exports.spawn (child_process.js:378:9)
    at /usr/lib/node_modules/pm2/lib/TreeKill.js:33:14
    at buildProcessTree (/usr/lib/node_modules/pm2/lib/TreeKill.js:79:12)
    at module.exports (/usr/lib/node_modules/pm2/lib/TreeKill.js:32:5)
    at Object.God.killProcess (/usr/lib/node_modules/pm2/lib/God/Methods.js:223:7)
    at Object.God.stopProcessId (/usr/lib/node_modules/pm2/lib/God/ActionMethods.js:265:9)
    at Object.God.restartProcessId (/usr/lib/node_modules/pm2/lib/God/ActionMethods.js:362:11)
    at /usr/lib/node_modules/pm2/lib/God/ActionMethods.js:395:20
2017-04-17 11:50:42: ===============================================================================
2017-04-17 11:50:42: [PM2] Resurrecting PM2
2017-04-17 11:50:44: App [app5] with id [11] and pid [18813], exited with code [1] via signal [SIGINT]
2017-04-17 11:50:44: App [app4] with id [7] and pid [18860], exited with code [0] via signal [SIGABRT]
2017-04-17 11:50:44: Starting execution sequence in -fork mode- for app name:app5 id:11
2017-04-17 11:50:44: App name:app5 id:11 online
module.js:471
    throw err;
    ^

Error: Cannot find module '/undefined'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.runMain (module.js:604:10)
    at run (bootstrap_node.js:393:7)
    at startup (bootstrap_node.js:150:9)
    at bootstrap_node.js:508:3
2017-04-17 11:50:44: Change detected on path core for app app6 - restarting
2017-04-17 11:50:44: Stopping app:app6 id:10
2017-04-17 11:50:45: App [app6] with id [10] and pid [18646], exited with code [0] via signal [SIGABRT]
2017-04-17 11:50:45: PM2 successfully forked
2017-04-17 11:50:47: ===============================================================================
2017-04-17 11:50:47: --- New PM2 Daemon started ----------------------------------------------------
2017-04-17 11:50:47: Time                 : Mon Apr 17 2017 11:50:47 GMT+0300 (MSK)
2017-04-17 11:50:47: PM2 version          : 2.4.4
2017-04-17 11:50:47: Node.js version      : 6.10.2
2017-04-17 11:50:47: Current arch         : x64
2017-04-17 11:50:47: PM2 home             : /root/.pm2
2017-04-17 11:50:47: PM2 PID file         : /root/.pm2/pm2.pid
2017-04-17 11:50:47: RPC socket file      : /root/.pm2/rpc.sock
2017-04-17 11:50:47: BUS socket file      : /root/.pm2/pub.sock
2017-04-17 11:50:47: Application log path : /root/.pm2/logs
2017-04-17 11:50:47: Process dump file    : /root/.pm2/dump.pm2
2017-04-17 11:50:47: Concurrent actions   : 2
2017-04-17 11:50:47: SIGTERM timeout      : 1600
2017-04-17 11:50:47: ===============================================================================
2017-04-17 11:50:47: Starting execution sequence in -fork mode- for app name:pm2-logrotate id:0
2017-04-17 11:50:47: App name:pm2-logrotate id:0 online
2017-04-17 11:50:47: Stopping app:pm2-logrotate id:0
2017-04-17 11:50:47: App [pm2-logrotate] with id [0] and pid [19836], exited with code [0] via signal [SIGINT]
2017-04-17 11:50:47: [HandleExit] PM2 is being killed, stopping restart procedure...
2017-04-17 11:50:47: pid=19836 msg=process killed
2017-04-17 11:50:47: ===============================================================================
2017-04-17 11:50:47: --- Stopping PM2 --------------------------------------------------------------
2017-04-17 11:50:47: Time                 : Mon Apr 17 2017 11:50:47 GMT+0300 (MSK)
2017-04-17 11:50:47: ===============================================================================
2017-04-17 11:50:47: RPC closed
2017-04-17 11:50:47: PUB closed
2017-04-17 12:14:43: ===============================================================================
2017-04-17 12:14:43: --- New PM2 Daemon started ----------------------------------------------------
2017-04-17 12:14:43: Time                 : Mon Apr 17 2017 12:14:43 GMT+0300 (MSK)
2017-04-17 12:14:43: PM2 version          : 2.4.4
2017-04-17 12:14:43: Node.js version      : 6.10.2
2017-04-17 12:14:43: Current arch         : x64
2017-04-17 12:14:43: PM2 home             : /root/.pm2
2017-04-17 12:14:43: PM2 PID file         : /root/.pm2/pm2.pid
2017-04-17 12:14:43: RPC socket file      : /root/.pm2/rpc.sock
2017-04-17 12:14:43: BUS socket file      : /root/.pm2/pub.sock
2017-04-17 12:14:43: Application log path : /root/.pm2/logs
2017-04-17 12:14:43: Process dump file    : /root/.pm2/dump.pm2
2017-04-17 12:14:43: Concurrent actions   : 2
2017-04-17 12:14:43: SIGTERM timeout      : 1600
2017-04-17 12:14:43: ===============================================================================
2017-04-17 12:14:43: Starting execution sequence in -fork mode- for app name:pm2-logrotate id:0
2017-04-17 12:14:43: App name:pm2-logrotate id:0 online
```

**Comment by vmarchaud** [comment]
ENOMEM = Out of memory, you need to free memory to let PM2 spawn more process.

**Comment by extensionsapp** [comment]
@vmarchaud Hello.
It's great that we now know the reason for the fall. But why after reboot all processes are not restored?
I did start each process using `pm2 save`
And I already wrote about this - https://github.com/Unitech/pm2/issues/2827 When pm2 is fall, for some reason it can not start the processes again. Such a feeling that immediately after the fall, he forgets that the command `pm2 save` was being made. If it does not work `pm2 save`, then why is it needed?
